### PR TITLE
added controller method and route for reply upvote

### LINKED ISSUE #159: [FEATURE REQUEST] : Create controller method to upvote a reply
The purpose of this feature is to give users the ability to upvote replies.

"RepliesController - PATCH /comments/{commentId}/replies/{replyId}/votes/upvote:
Create controller method to upvote a reply"

### PR DESCRIPTION
**Pull request by Ochuwa-sophie**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**Linting errors have not been fixed because I am getting an error I am uncertain of it's origin.**
**What does this PR do?**
**Added the controller method and route for the reply upvote.**
**Fixes #159**  

Summarise the main tasks handled in the pull request

**description of Task to be completed?**

Describe (in detail) what the task completed in the pull request does as per the relevant task

**How should this be manually tested?**

Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

**Any background context you want to provide?**

Any pertinent information that should be considered
None
**What is the link to the issue on Github?**

[Issue #159 ](https://github.com/microapidev/comment-microapi/issues/159)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here
**None**

### CODE REVIEW

File: controller/repliesController.js [hunk 1]
```diff
@@ -159,104 +162,73 @@ const getReplyVotes = async (req, res, next) => {
...
+
+        //get total number of votes
+        const totalVotes = totalUpVotes + totalDownVotes;
+
+        const data = {
+          commentId: comment._id,
+          numOfVotes: totalVotes,
+          numOfUpVotes: totalUpVotes,
+          numOfDownVotes: totalDownVotes,
+        };
```
Comment (simply-alliv) [inline controller/repliesController.js:213]: Please add the replyId too. Also you already have the commentId from the params. It's not necessary for you to fetch the _id of the comment :
```suggestion
        const data = {
          replyId,
          commentId,
          numOfVotes: totalVotes,
          numOfUpVotes: totalUpVotes,
          numOfDownVotes: totalDownVotes,
        };
```

File: controller/repliesController.js [hunk 2]
```diff
@@ -132,20 +132,23 @@ const createReply = async (req, res, next) => {
...
+        if (!ObjectId.isValid(replyId)) {
+          return next(new CustomError(404, "Invalid ID"));
+          }
+      
+
+    const parentComment = await Comments.findById({ _id: commentId
+    });
 
     // Check to see if the parent comment exists.
     if (!parentComment) {
```
Comment (simply-alliv) [inline controller/repliesController.js:154]: What's happening here? I think that you are missing some code. If you tested your code then this would have been flagged.

Comment (Ochuwa-sophie) [reply controller/repliesController.js:154]: Got it. Will work on these

File: controller/repliesController.js [hunk 3]
```diff
@@ -159,104 +162,74 @@ const getReplyVotes = async (req, res, next) => {
...
-      replyId: reply._id,
-      commentId: reply.commentId,
-      numOfFlags: reply.flags.length,
-    };
-
-    return responseHandler(
-      res,
-      200,
-      data,
+
```
Comment (dave-ok) [inline controller/repliesController.js:230]: Why did you delete this entire section of code

File: controller/repliesController.js [hunk 4]
```diff
@@ -159,104 +162,74 @@ const getReplyVotes = async (req, res, next) => {
...
     const reply = await Replies.findById(replyId);
 
-    // Check to see if the reply exists.
+// Check to see if the reply exists.
     if (!reply) {
       return next(
         new CustomError(
           404,
-          `Reply with the ID ${replyId} does not exist or has been deleted`
+          'Reply with the ID ${replyId} does not exist or has been deleted'
```
Comment (dave-ok) [inline controller/repliesController.js:170]: Change this back to bacticks (`) instead of single quotes(') for interpolation to work.

Comment (Ochuwa-sophie) [reply controller/repliesController.js:170]: Done
